### DISCUSSION TITLE
Fail NX build for an NX deprecation warning

### DISCUSSION THREAD

**Discussion by juddgaddie** [nrwl/nx#23417] [Feature Requests]
## Description
I NX configuration option that will result in the build failing if a NX deprecation warning exits e.g. https://nx.dev/deprecated/global-implicit-dependencies#global-implicit-dependencies

## Motivation
So deprecations are not accidentally introduced without anyone noticing.

**Comment by juddgaddie** [comment]
Is this going anywhere ?

**Comment by FrozenPandaz** [comment]
Could you please explain more the scenarios where deprecated options are being added unintentionally?
